File: examples/eApi v1.9/java/eAPI-v19-connector/src/main/java/cz/monetplus/mips/eapi/v19/connector/entity/MallpayLogisticsRequest.java
package cz.monetplus.mips.eapi.v19.connector.entity;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;

@EqualsAndHashCode(callSuper = false)
@Data
@NoArgsConstructor @AllArgsConstructor
public class MallpayLogisticsRequest  extends SignBase {
    private String merchantId;
    private String payId;
    private String event;
    private String date;
    private MallpayOrderRef fulfilled;
    private MallpayOrderRef cancelled;
    private String deliveryTrackingNumber;

    @Override
    public String toSign() {
        StringBuilder sb = new StringBuilder();
        add(sb, merchantId);
        add(sb, payId);
        add(sb, event);
        add(sb, date);
        if(fulfilled != null) add(sb, fulfilled.toSign());
        if(cancelled != null) add(sb, cancelled.toSign());
        add(sb, deliveryTrackingNumber);
        add(sb, dttm);
        deleteLast(sb);
        return sb.toString();
    }
}

File: examples/eApi v1.9/dotNET/CsobGTWDemo/CsobGatewayClientExample/Communication/DataObjects/MallpayLogisticsRequest.cs
using System.Text;
using Newtonsoft.Json;

namespace CsobGatewayClientExample.Communication.DataObjects;

public class MallpayLogisticsRequest : SignBaseRequest
{
    [JsonProperty("payId")] public string? PayId { get; set; }

    [JsonProperty("date")] public string? Date { get; set; }

    [JsonProperty("fulfilled")] public MallpayOrderRef? Fulfilled { get; set; }

    [JsonProperty("cancelled")] public MallpayOrderRef? Cancelled { get; set; }

    [JsonProperty("deliveryTrackingNumber")]
    public string? DeliveryTrackingNumber { get; set; }

    [JsonProperty("event")] public string? Event { get; set; }


    public override string ToSign()
    {
        var sb = new StringBuilder();
        Add(sb, MerchantId);
        Add(sb, PayId);
        Add(sb, Event);
        Add(sb, Date);
        Add(sb, Fulfilled?.ToSign());
        Add(sb, Cancelled?.ToSign());
        Add(sb, DeliveryTrackingNumber);
        Add(sb, Dttm);
        DeleteLast(sb);
        return sb.ToString();
    }
}